### ISSUE TITLE
Change diagnostics "underline" to distinguish between squiggly and straight

### ISSUE THREAD

**Issue by peterkos** [sublimelsp/LSP#1130]
One of the things that bothers me is how `highlight` can distinguish between `squiggly` and `underline`, but for diagnostics, it's `underline` which is **actually** squiggly:
<img width="396" alt="Screen Shot 2020-06-15 at 7 48 56 PM" src="https://user-images.githubusercontent.com/2782749/84716493-409d3100-af41-11ea-9844-e8657e06ae70.png">

```
// [LSP.sublime-settings]
// Highlighting style of code diagnostics.
// Valid values are "underline" or "box"
"diagnostics_highlight_style": "underline",
```

It'd be amazing if there was an ability to make diagnostics be solid underlines, not squiggly underlines. I've tried poking through the source code but python is a bit hard to navigate imo.

**Comment by peterkos** [comment]
Update: I found where the `highlight` code sets its line configuration:
https://github.com/sublimelsp/LSP/blob/10c4004be55dd54ff61473093f03e2a90ca867ca/plugin/highlights.py#L106-L113
If this issue is accepted the logic in `diagnostics.py` would probably have to be updated to match that as it currently just uses a static list for flags + an inline check to switch to `box_flags`:
https://github.com/sublimelsp/LSP/blob/10c4004be55dd54ff61473093f03e2a90ca867ca/plugin/diagnostics.py#L31-L32
https://github.com/sublimelsp/LSP/blob/10c4004be55dd54ff61473093f03e2a90ca867ca/plugin/diagnostics.py#L260-L262

**Comment by rchl** [comment]
I think that straight underline is used to mark matching symbols right now (when you put the caret at some variable name for example. So that would probably conflict.

**Comment by peterkos** [comment]
It can already conflict with the existing diagnostic style if the user selects symbol matching to be `squiggly`, and same if they opt in to `box` for both of those.
Either way it would be more consistent both in the codebase (flag matching), and in the UI to allow for the same customization options for different highlight styles by LSP. (That, and popular libraries such as SublimeLinter allow for this customization! A lot of users probably are coming to LSP from there)

**Comment by rwols** [comment]
Fixed in #1170. Both diagnostics_highlight_style and diagnostics_highlight_style work exactly the same.
